server/src/main/java/org/opensearch/cluster/coordination/JoinTaskExecutor.java

```diff
@@ -160,6 +175,10 @@ public ClusterTasksResult<Task> execute(ClusterState currentState, List<Task> jo
         }
 
         DiscoveryNodes.Builder nodesBuilder = DiscoveryNodes.builder(newState.nodes());
+        RepositoriesMetadata repositoriesMetadata = remoteStoreNodeService.updateRepositoriesMetadata(
+            joiningNodes.stream().filter(task -> task.node() != null).findAny(),
+            currentState.getMetadata().custom(RepositoriesMetadata.TYPE)
+        );
```

Comment:
In case you picked the incompatible node here in findAny(), then it will update the incompatible repo as below in the JoiningNodes forloop it will catch the IllegalStateException and move on to process next node.

Reply:
can you pick up repos from existingState?

Reply:
To keep code simple, like you are picking the existingNodes.get(0) for compatibility check, you can use that for repo registration as well to keep the code consistent. When mixed mode is introduced later, you can make appropriate changes.